### PR TITLE
KOGITO-4198 - Add jit dmnresult endpoint and refactoring

### PR DESCRIPTION
**Pull request by r00ta**
JIRA: https://issues.redhat.com/browse/KOGITO-4198

This PR aims to add the endpoint `/jitdmn/dmnresult` to retrieve the entire dmn result that includes messages and metadata of the execution.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: jitexecutor/jitexecutor-dmn/src/main/resources/META-INF/native-image/org.kie.kogito/jitexecutor-dmn/reflect-config.json [hunk 1]
```diff
@@ -29,1210 +29,1228 @@
...
-      "allPublicFields" : true
-    },
-    {
-      "name":"sun.misc.Unsafe",
-      "fields":[{"name":"theUnsafe"}],
-      "methods":[{"name":"allocateInstance","parameterTypes":["java.lang.Class"] }]
-    }
-  ]
\ No newline at end of file
+  "name":"com.thoughtworks.xstream.converters.ConverterLookup",
```
Comment (tarilabs) [inline jitexecutor/jitexecutor-dmn/src/main/resources/META-INF/native-image/org.kie.kogito/jitexecutor-dmn/reflect-config.json]: @r00ta there is a "giant" modification of ~2432 lines on this file which I'm not sure why is needed.
Can you kindly clarify either what needs change in this file, or revert any modification on this file, please?

File: jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/JITDMNResource.java [hunk 2]
```diff
@@ -14,51 +14,55 @@
...
-        return result;
+        return Response.ok(restResulk).build();
+    }
+
+    @POST
+    @Path("/dmnresult")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response jitdmnResult(JITDMNPayload payload) {
+        JITDMNEvaluationResult evaluationResult = jitdmnService.evaluateModel(payload.getModel(), payload.getContext());
```
Comment (tarilabs) [inline jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/JITDMNResource.java]: Rather then introducing yet another DTO (`JITDMNEvaluationResult `) I would re-use the one already available on Kogito, which is structurally equivalent to the one inherited to kie-server.
I believe it contains all the needs/fields which seems to have been introduced here?

File: jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/SchemaResource.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.eclipse.microprofile.openapi.OASFactory;
+import org.eclipse.microprofile.openapi.models.OpenAPI;
+import org.kie.kogito.jitexecutor.dmn.JITDMNService;
+
+@Path("jitdmn/schema")
+public class SchemaResource {
```
Comment (tarilabs) [inline jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/SchemaResource.java:46]: I am not sure this refactoring is worthy, I mean to separate "service"Vs"rest resource" for the JSONSchema / OAS.
At the end of the day, the way the OAS is built is tied to the API being exposed via REST, so this separation imho is a bit premature/superflous.
I am not strongly opinionated against, I am just concerned about overrefactoring early ;)

**Review by tarilabs — CHANGES_REQUESTED** [review]
Setting as "request changes" mainly only for the note about `reflect-config.json`.
The other comment optionals
Thanks!

File: jitexecutor/jitexecutor-dmn/src/main/resources/META-INF/native-image/org.kie.kogito/jitexecutor-dmn/reflect-config.json (hunk 1 above)
Comment (r00ta) [reply jitexecutor/jitexecutor-dmn/src/main/resources/META-INF/native-image/org.kie.kogito/jitexecutor-dmn/reflect-config.json]: Hi @tarilabs , this json file is not indented properly and my editor refactored it automatically. I've reverted the changes

File: jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/JITDMNResource.java (hunk 2 above)
Comment (r00ta) [reply jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/JITDMNResource.java]: You mean this one? https://github.com/kiegroup/drools/blob/24a0abbc754733e7d4d7db7fbe49fc6d4fb323e8/kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/internal/utils/DMNEvaluationUtils.java#L110

File: jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/SchemaResource.java (hunk 3 above)
Comment (r00ta) [reply jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/SchemaResource.java:46]: fair enough for me. Let's refactor it when unit tests will be implemented as well as exception handling. Reverted

File: jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/JITDMNResource.java (hunk 2 above)
Comment (tarilabs) [reply jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/JITDMNResource.java]: No,
I mean this one: https://github.com/kiegroup/kogito-runtimes/blob/f8ad6324c8a7120f3fd7164adce1849cf8dc5368/drools/kogito-dmn/src/main/java/org/kie/kogito/dmn/rest/DMNResult.java#L36

which IIRC is the one directly returned by the `/dmnresult` also for common REST

File: jitexecutor/jitexecutor-dmn/src/main/resources/META-INF/native-image/org.kie.kogito/jitexecutor-dmn/reflect-config.json [hunk 4]
```diff
@@ -1230,6 +1230,24 @@
...
+    },
+    {
+      "name":"org.kie.kogito.dmn.rest.DMNDecisionResultSQ",
+      "allDeclaredConstructors" : true,
+      "allPublicConstructors" : true,
+      "allDeclaredMethods" : true,
+      "allPublicMethods" : true,
+      "allDeclaredFields" : true,
+      "allPublicFields" : true
+    },
```
Comment (tarilabs) [inline jitexecutor/jitexecutor-dmn/src/main/resources/META-INF/native-image/org.kie.kogito/jitexecutor-dmn/reflect-config.json:1250]: thanks for having applied only the actual mods :)

**Review by tarilabs — APPROVED** [review]

**Review by danielezonca — APPROVED** [review]

File: jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/JITDMNResource.java (hunk 2 above)
Comment (danielezonca) [reply jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/JITDMNResource.java]: I agree that if it is possible it is better to reuse that dto

Comment (r00ta) [reply jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/JITDMNResource.java]: Done

**Comment by r00ta** [comment]
Hi @tarilabs , let me know if we are good to go

**Review by hellowdan — APPROVED** [review]

**Pull request merged**